packages/astro/src/core/build/static-build.ts

```diff
@@ -374,7 +374,10 @@ async function cleanStaticOutput(
 ) {
 	const allStaticFiles = new Set();
 	for (const pageData of eachPageData(internals)) {
-		if (pageData.route.prerender && !pageData.hasSharedModules) {
+		if (
+			pageData.route.prerender &&
+			!(pageData.hasSharedModules || pageData.hasAstroVirtualModule)
```

Comment:
is this part the actual fix of the issue? Can you explain why this is needed if so?

Reply:
Yes this is part of the fix. This is where we collect the chunks, few lines below we clean them up by making them no-op:

https://github.com/withastro/astro/blob/317d18ef8c9cf4fd13647518e3fd352774a86481/packages/astro/src/core/build/static-build.ts#L397-L413